app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,24 @@
+class CommentsController < ApplicationController
+  def new
+    @comment = Comment.new
+  end
+
+  def create
+    @comment = Comment.new(comment_params)
+    @post = Post.find(params[:post_id])
+    @comment.author_id = current_user.id
+    @comment.post_id = @post.id
+
+    if @comment.save
+      redirect_to user_post_path(user_id: @post.author_id, id: @post.id)
+    else
+      render :new, alert: 'An error has occurred while creating the comment'
+    end
+  end
```

Comment:
✅ A comment is created on behalf of the currently logged in user (current_user) of the app 👍🏻 

Kindly refactor the create action to 

```ruby
def create
    @comment = @current_user.comments.new(params)
    @comment.post_id = params[:post_id]

    if @comment.save
      redirect_to post_path(@comment.post)
    else
      @article = Post.find(params[:post_id])
      redirect_to post_path(@post)
    end
  end
```